test/Api.Test/SecretsManager/Controllers/ServiceAccountsControllerTests.cs

```diff
@@ -90,12 +95,70 @@ await sutProvider.GetDependency<ICreateServiceAccountCommand>().DidNotReceiveWit
 
     [Theory]
     [BitAutoData]
-    public async void CreateServiceAccount_Success(SutProvider<ServiceAccountsController> sutProvider,
+    public async void CreateServiceAccount_OrgDoesntExist_Throws(SutProvider<ServiceAccountsController> sutProvider,
         ServiceAccountCreateRequestModel data, Guid organizationId)
     {
         sutProvider.GetDependency<IAuthorizationService>()
             .AuthorizeAsync(Arg.Any<ClaimsPrincipal>(), data.ToServiceAccount(organizationId),
                 Arg.Any<IEnumerable<IAuthorizationRequirement>>()).ReturnsForAnyArgs(AuthorizationResult.Success());
+
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(Arg.Is(organizationId)).ReturnsNull();
+        sutProvider.GetDependency<IUserService>().GetProperUserId(default).ReturnsForAnyArgs(Guid.NewGuid());
+        var resultServiceAccount = data.ToServiceAccount(organizationId);
+        sutProvider.GetDependency<ICreateServiceAccountCommand>().CreateAsync(default, default)
+            .ReturnsForAnyArgs(resultServiceAccount);
+
+        await Assert.ThrowsAsync<NotFoundException>(() => sutProvider.Sut.CreateAsync(organizationId, data));
+        await sutProvider.GetDependency<ICreateServiceAccountCommand>().DidNotReceiveWithAnyArgs()
+            .CreateAsync(Arg.Any<ServiceAccount>(), Arg.Any<Guid>());
+    }
+
+    [Theory]
+    [BitAutoData(0)]
+    [BitAutoData(1)]
+    [BitAutoData(2)]
+    public async void CreateServiceAccount_NewServiceAccountSlotsRequired_CheckCallsUpdate(int newSlotsRequired, SutProvider<ServiceAccountsController> sutProvider,
+        ServiceAccountCreateRequestModel data, Guid organizationId, Organization mockOrg)
+    {
+        mockOrg.Id = organizationId;
+        sutProvider.GetDependency<IAuthorizationService>()
+            .AuthorizeAsync(Arg.Any<ClaimsPrincipal>(), data.ToServiceAccount(organizationId),
+                Arg.Any<IEnumerable<IAuthorizationRequirement>>()).ReturnsForAnyArgs(AuthorizationResult.Success());
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(Arg.Is(organizationId)).Returns(mockOrg);
+        sutProvider.GetDependency<ICountNewServiceAccountSlotsRequiredQuery>()
+            .CountNewServiceAccountSlotsRequiredAsync(Arg.Is(organizationId), Arg.Is(1))
```

Comment:
No need to use argument matchers in this case:

```suggestion
            .CountNewServiceAccountSlotsRequiredAsync(organizationId, 1)
```

Reply:
Resolved in https://github.com/bitwarden/server/pull/3093/commits/0f3beda47fb78eddc0a67108e3f6922345c93dc5